hyperspy/axes.py

```diff
@@ -57,13 +63,142 @@ def generate_axis(offset, scale, size, offset_index=0):
+        else:
+            return scale, offset, units
+
+    def convert_to_units(self, units=None, inplace=True, factor=0.25):
+        """ Convert the scale and the units of the current axis. If the unit 
+        of measure is not supported by the pint library, the scale and units 
+        are not changed.
+
+        Parameters
+        ----------
+        units : {str | None}
+            Default = None
+            If str, the axis will be converted to the provided units.
+            If `None`, the scale and the units are converted to the appropriate 
+            scale and units to avoid displaying scalebar with >3 digits or too 
+            small number. This can be tweaked by the `factor` argument.
+        inplace : bool
+            If `True`, convert the axis in place. if `False` return the  
+            `scale`, `offset` and `units`.
+        factor : float
+            'factor' an adjustable value used to determine the prefix of the 
+            units.
+        """
+        if units is None:
+            out = self._convert_compact_units(factor, inplace=inplace)
+        else:
+            out = self._convert_units(units, inplace=inplace)
+        return out
+
+    def _get_quantity(self, attribute='scale'):
+        if attribute == 'scale' or attribute == 'offset':
+            units = self.units
+            if units == t.Undefined:
+                units = ''
+            return self.__dict__[attribute] * _ureg(units)
+        else:
+            raise ValueError('`attribute` argument can only take the `scale` '
+                             'or the `offset` value.')
+
+    def _set_quantity(self, value, attribute='scale'):
+        if attribute == 'scale' or attribute == 'offset':
+            units = '' if self.units == t.Undefined else self.units
+            if isinstance(value, str):
+                value = _ureg.parse_expression(value)
+            if isinstance(value, float):
+                value = value * _ureg(units)
+
+            # to be consistent, we also need to convert the other one
+            # (scale or offset) when both units differ.
+            if value.units != units and value.units != '' and units != '':
+                other = 'offset' if attribute == 'scale' else 'scale'
+                other_quantity = self._get_quantity(other).to(value.units)
+                self.__dict__[other] = float(other_quantity.magnitude)
+
+            self.units = '{:~}'.format(value.units)
+            self.__dict__[attribute] = float(value.magnitude)
+        else:
+            raise ValueError('`attribute` argument can only take the `scale` '
+                             'or the `offset` value.')
+
+    @property
+    def units(self):
+        return self._units
+
+    @units.setter
+    def units(self, s):
+        if s == t.Undefined:
+            self._units = s
+        elif s == '':
+            self._units = t.Undefined
+        else:
+            self._units = s.replace('um', 'µm').replace(' ', '')
```

Comment:
I have opened an issue in the [pint repo](https://github.com/hgrecco/pint/issues/666) to ask. Pint seems to have a fairly good unicode support, so I don't see, why it would not work. Maybe, to avoid issue with python < 2.6?

Reply:
Personally I would prefer not to perform this translation in HyperSpy as it can lead to unexpected issues. For example, somebody may define an unit that contains the "um" chain. Also, it is not so hard to get write μ even I just did it!

Reply:
Yes, indeed, this is not right, I revert it.